altitude_estimators:

  # The available altitude estimators
  altitude_estimators: [
    "HEIGHT",
    "PLANE",
    "BRICK",
    "VIO",
    "ALOAM",
    "BARO",
    "ALOAMGARM"
  ] 

  # The measured states of the model
  model_states: [
    "HEIGHT", 
    "VELOCITY", 
    "ACCELERATION"
  ] 

  # The available measurements
  measurements: [
    "height_range",
    "height_sonar",
    "height_plane",
    "height_brick",
    "height_vio",
    "height_aloam",
    "height_baro",
    "vel_baro",
    "vel_vio",
    "acc_imu"
  ]

  # The fused measurements for each state estimator
  fused_measurements:
    HEIGHT: ["height_range", "vel_baro"]
    PLANE: ["height_plane", "vel_baro"]
    BRICK: ["height_brick", "vel_baro"]
    VIO: ["height_vio", "vel_vio"]
    ALOAM: ["height_aloam", "vel_baro"]
    BARO: ["height_baro", "vel_baro"]
    ALOAMGARM: ["height_aloam", "height_range", "vel_baro"]
    # ALOAMGARM: ["height_aloam", "vel_baro"]

  # Mapping of measurements to states
  measurement_states:
    height_range: HEIGHT
    height_sonar: HEIGHT
    height_plane: HEIGHT
    height_brick: HEIGHT
    height_vio: HEIGHT
    height_aloam: HEIGHT
    height_baro: HEIGHT
    vel_baro: VELOCITY
    vel_vio: VELOCITY
    acc_imu: ACCELERATION

  # Mapping states to model
  state_mapping:
    HEIGHT: [1.0, 0.0, 0.0]
    VELOCITY: [0.0, 1.0, 0.0]
    ACCELERATION: [0.0, 0.0, 1.0]

# Untilted garmin filter (true height above terrain unsuitable for control)
height:
  Q: [0.1]
  R: [0.01]
